packages/authentication/src/providers/google.ts

```diff
@@ -0,0 +1,275 @@
+import {
+  AbstractAuthenticationModuleProvider,
+  AuthenticationResponse,
+} from "@medusajs/types"
+import { AuthProviderService, AuthUserService } from "@services"
+
+import { OAuth2 } from "oauth"
+import url, { UrlWithParsedQuery } from "url"
+import jwt, { JwtPayload } from "jsonwebtoken"
+import { MedusaError } from "@medusajs/utils"
+import { TreeLevelColumn } from "typeorm"
+
+type InjectedDependencies = {
+  authUserService: AuthUserService
+  authProviderService: AuthProviderService
+}
+
+type GoogleProviderConfig = {
+  callbackURL: string
+  clientID: string
+  clientSecret: string
+}
+
+class GoogleProvider extends AbstractAuthenticationModuleProvider {
+  public static PROVIDER = "google"
+  public static DISPLAY_NAME = "Google Authentication"
+
+  private authorizationUrl_ = "https://accounts.google.com/o/oauth2/v2/auth"
+  private tokenUrl_ = "https://www.googleapis.com/oauth2/v4/token"
+
+  protected readonly authUserSerivce_: AuthUserService
+  protected readonly authProviderService_: AuthProviderService
+
+  constructor({ authUserService, authProviderService }: InjectedDependencies) {
+    super()
+
+    this.authUserSerivce_ = authUserService
+    this.authProviderService_ = authProviderService
+  }
+
+  private async validateConfig(config: Partial<GoogleProviderConfig>) {
+    if (!config.clientID) {
+      throw new Error("Google clientID is required")
+    }
+
+    if (!config.clientSecret) {
+      throw new Error("Google clientSecret is required")
+    }
+
+    if (!config.callbackURL) {
+      throw new Error("Google callbackUrl is required")
+    }
+  }
+
+  private originalURL(req) {
```

Comment:
todo: type the req with the required properties etc. Also, this could be a static utils on the abstract class